library/core/src/slice/raw.rs

```diff
@@ -179,15 +181,16 @@ pub const fn from_mut<T>(s: &mut T) -> &mut [T] {
 ///   the last element, such that the offset from the end to the start pointer is
 ///   the length of the slice.
 ///
-/// * The range must contain `N` consecutive properly initialized values of type `T`:
+/// * The entire memory range of this slice must be contained within a single allocated object!
+///   Slices can never span across multiple allocated objects.
```

Comment:
I moved this to the top level since having it as a sub-clause of "`N` consecutive properly initialized values of type `T`" doesn't really make sense IMO.